### PR TITLE
1372: Rtl text on map

### LINKED ISSUE #2295: IGAPP-1372: Support RTL for POI Title on Map
# f1sh1918 - 7.8.2023, 10:08:48

As a user with a rtl language i would like to have the title of the pois displayed on the map

**Acceptance Criteria**
* Display the title of the poi in rtl language

 

Note:
* add arabic font glyphs to the mapstyles repo

https://github.com/digitalfabrik/integreat-maplibre-style

 

 

**Environment**: -
**Linked issues:**

### PR DESCRIPTION
**Pull request by f1sh1918**
### Short description

With this pr we enable rtl support on the map canvas, that means we can display f.e. texts with arabic languages on the map.
In this case we can display the poi title in arabic, that can be found here

http://localhost:9000/testumgebung/de/locations/cafe-mit-%D9%85%D9%82%D9%87%D9%89

image

### Proposed changes
- created font-glyphs for map canvas that support arabic (https://github.com/digitalfabrik/integreat-maplibre-style/pull/17)
- adjusted style.json to use fonts including the new font-glyphs (https://github.com/digitalfabrik/integreat-maplibre-style/pull/16)
- add rtl plugin to our repository and override standard plugin url
- adjust web & native map layers to use `Noto Sans Regular` which supports arabic languages

### Side effects

- adjust web & native map layers to use `Noto Sans Regular` which supports arabic languages

### Resolved issues

resolves #2295

Tested on chrome, ios and android

### Note
- now we could get poi titles translated into different languages from the cms which is currently restricted

image

### CODE REVIEW

File: web/src/components/MapView.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import * as mapLibreGl from 'maplibre-gl'
+import maplibregl from 'maplibre-gl'
```
Comment (f1sh1918) [inline web/src/components/MapView.tsx:1]: the exposed interfaces didn't work properly f.e. the `setRTLPlugin` couldn't be found in ts-check

File: web/src/components/MapView.tsx [hunk 2]
```diff
@@ -81,15 +82,21 @@ const MapView = forwardRef(
...
     const [cursor, setCursor] = useState<MapCursorType>('auto')
-    const [mapRef, setMapRef] = useState<mapLibreGl.Map | null>(null)
+    const [mapRef, setMapRef] = useState<maplibregl.Map | null>(null)
     const theme = useTheme()
 
     const { viewportSmall, height } = useWindowDimensions()
 
+    useEffect(() => {
+      if (maplibregl.getRTLTextPluginStatus() === 'unavailable') {
+        maplibregl.setRTLTextPlugin(mapConfig.rtlPluginUrl, err => err && reportError(err), true)
```
Comment (f1sh1918) [inline web/src/components/MapView.tsx:92]: had to avoid sending `null` in the callback, since they are optional

**Review by LeandraH — APPROVED** [review]
Cool! Tested on Firefox and real iOS, looks great!

**Review by steffenkleinle — APPROVED** [review]
Tested on firefox and web, well done : tada:

**Pull request merged**